### ISSUE TITLE
New rule proposal: Disallow calling of a known spy object created in the same spec

### ISSUE THREAD

**Issue by KevinPartingtonTR** [tlvince/eslint-plugin-jasmine#364]
**Is your feature request related to a problem? Please describe.**

I'm seeing developers in my codebase do the following absolute nonsense:

```js
describe('a suite', () => {
  it('a spec', () => {
    // arrange
    const objUnderTest = createObjectUnderTest();
    spyOn(objUnderTest, 'someMethod');

    // act
    objUnderTest.someMethod();

    // assert
    expect(objUnderTest.someMethod).toHaveBeenCalled();
  });
});
```

**Describe the solution you'd like**

I would like a rule which checks, within a spec, that an object/property pair that are passed to a `spyOn` call are not then called as a function within the same spec.

**Describe alternatives you've considered**

Firing my developers hasn't been an option, but I'm working on that...

I could create a custom rule pretty easily (and I probably will), but I wanted to suggest it for eslint-plugin-jasmine since it seems like a use case people would universally want to prevent.

**Additional context**

N/A
